src/sentry/integrations/aws_lambda/integration.py

```diff
@@ -347,37 +349,50 @@ def dispatch(self, request, pipeline):
         lambda_functions = get_supported_functions(lambda_client)
         lambda_functions.sort(key=lambda x: x["FunctionName"].lower())
 
-        failures = []
-        success_count = 0
-
-        for function in lambda_functions:
+        def is_lambda_enabled(function):
             name = function["FunctionName"]
             # check to see if the user wants to enable this function
-            if not enabled_lambdas.get(name):
-                continue
+            return enabled_lambdas.get(name)
 
+        lambda_functions = filter(is_lambda_enabled, lambda_functions)
+
+        def _enable_lambda(function):
             try:
                 enable_single_lambda(lambda_client, function, sentry_project_dsn)
-                success_count += 1
+                return (True, function, None)
             except Exception as e:
-                # need to make sure we catch any error to continue to the next function
-                err_message = six.text_type(e)
-                invalid_layer = get_invalid_layer_name(err_message)
-                if invalid_layer:
-                    err_message = _(INVALID_LAYER_TEXT) % invalid_layer
+                return (False, function, e)
+
+        failures = []
+        success_count = 0
+
+        with ThreadPoolExecutor(max_workers=10) as _lambda_setup_thread_pool:
+            # use threading here to parallelize requests
+            for success, function, e in _lambda_setup_thread_pool.map(
```

Comment:
I don't think this code will catch a TimeoutError properly. The docs say `The returned iterator raises a concurrent.futures.TimeoutError if __next__() is called and the result isn’t available after timeout seconds from the original call to Executor.map().` I don't think the try/except you have in _enable_lambda will catch that.

Reply:
@evanh Unfortunately, I don't know if there is a good way to handle the `TimeoutError`. If that happens, I won't know which function it failed on since I normally get that from the return value of `_enable_lambda`. I won't be able to add the failures to `failures`. Maybe I don't need the timeout argument at all since if we get a request timeout, that will generate an error anyway. Thoughts?

cc @EvanPurkhiser

Reply:
> if we get a request timeout (not the thread timeout),

I agree we might not even need it, I was sort of alluding to that, I just didn't actually read the code to know weather or not it would time out deeper into the api :)

Reply:
yea, taking it out cause we don't need it